examples/vpc_peering_common_dual_stack/main.tf

```
@@ -0,0 +1,273 @@
+  )
+}
+
+resource "local_sensitive_file" "init_cfg" {
+
+  for_each = { for k, v in var.vmseries : k => v
+    if can(v.bootstrap_template_map)
+  }
+
+  filename = "files/${each.key}/config/init-cfg.txt"
+  content = templatefile(
+    "templates/init-cfg.tmpl",
+    { bootstrap_options = merge(var.vmseries_common.bootstrap_options, each.value.bootstrap_options) }
+  )
+}
+
+module "bootstrap" {
+  source = "../../modules/bootstrap"
+
+  for_each = var.bootstrap_buckets
+
+  folders = keys(var.vmseries)
+
+  name_prefix     = "${var.name_prefix}${each.value.bucket_name_prefix}"
+  service_account = module.iam_service_account[each.value.service_account_key].email
+  location        = each.value.location
+  files = merge(
+    { for k, v in var.vmseries : "files/${k}/config/bootstrap.xml" => "${k}/config/bootstrap.xml" if can(v.bootstrap_template_map) },
+    { for k, v in var.vmseries : "files/${k}/config/init-cfg.txt" => "${k}/config/init-cfg.txt" if can(v.bootstrap_template_map) },
+  )
+}
+
+module "vpc" {
+  source = "../../modules/vpc"
+
+  for_each = var.networks
+
+  project_id                      = var.project
+  name                            = "${var.name_prefix}${each.value.vpc_name}"
+  create_network                  = each.value.create_network
+  delete_default_routes_on_create = each.value.delete_default_routes_on_create
+  mtu                             = each.value.mtu
+  routing_mode                    = each.value.routing_mode
+  enable_ula_internal_ipv6        = try(each.value.enable_ula_internal_ipv6, false)
+  internal_ipv6_range             = try(each.value.internal_ipv6_range, null)
+  subnetworks = { for k, v in each.value.subnetworks : k => merge(v, {
+    name = "${var.name_prefix}${v.name}"
+    })
+  }
+  firewall_rules = try({ for k, v in each.value.firewall_rules : k => merge(v, {
+    name = "${var.name_prefix}${v.name}"
+    })
+  }, {})
+}
+
+resource "google_compute_route" "this" {
+
+  for_each = var.routes
+
+  name         = "${var.name_prefix}${each.value.name}"
+  dest_range   = each.value.destination_range
+  network      = module.vpc[each.value.vpc_network_key].network.self_link
+  next_hop_ilb = module.lb_internal[each.value.lb_internal_key].forwarding_rule
+  priority     = 100
+}
+
+resource "null_resource" "policy_routes" {
+  for_each = var.policy_routes
+
+  triggers = {
+    project = var.project
+    name    = "${var.name_prefix}${each.key}-pbr"
+  }
+  provisioner "local-exec" {
+    command = join(" ", [
+      "gcloud beta network-connectivity policy-based-routes create ${self.triggers.name}",
+      "--priority=90",
+      "--source-range=::/0",
+      "--destination-range=::/0",
+      "--ip-protocol=ALL",
+      "--protocol-version=IPv6",
+      "--network=${module.vpc[each.value.vpc_network_key].network.id}",
+      "--next-hop-ilb-ip=${split("/", module.lb_internal[each.value.lb_internal_key].address)[0]}",
+      "--description=Policy-based_route_for_IPv6_outbound_connectivity",
```

Why not put the description in single quotes and use spaces instead of underscores?

---

If use `'Policy-based route for IPv6 outbound connectivity'` `gcloud` will fail for Terraform under Windows. For whatever reason single quotes do NOT make their way to the shell and spaces in the `'Policy-based route for IPv6 outbound connectivity'` are treated as parameter separators breaking the command exectution

---

```suggestion
      "--description='Policy based route for IPv6 outbound connectivity'",
```
This works like a charm :) At least under some WSL.